### PR TITLE
Add rockspec file for building via LuaRocks

### PR DESCRIPTION
**Pull request by bzick**
LuaRocks supports [building a package through cmake](https://github.com/luarocks/luarocks/wiki/Rockspec-format#cmake). For that need to specify the build type in the rockspec file.

Also, now you may submit the package to [luarocks.org](https://luarocks.org/upload).

Readme updated.

### CODE REVIEW

**Comment by bzick** [comment]
Add rockspec for git (master) version. A new rockpec file must be created for each release/tag..

File: rockspec/lua-cbson-git-1.rockspec [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    summary = "Lua bindings for libbson",
+    homepage = "https://github.com/isage/lua-cbson",
+    maintainer = "isage.dna@gmail.com",
+}
+build = {
+    type = "cmake",
+    variables = {
+        LUA_INCLUDE_DIR = "$(LUA_INCDIR)",
+        LUA_LIBRARIES = "$(LUA_LIBDIR)",
+        LUA_LIBRARY = "$(LUA_LIBDIR)/lua",
```
Comment (bzick) [inline rockspec/lua-cbson-git-1.rockspec]: LUA_LIBDIR has already required path e.g.   `/usr/local/lib`. But cmake or luarocks trim one level to `/usr/local` in `LUA_LIBRARY`. Suffix `/lua` added for this specific behavior and has no sense. But it works (tested on linux and macos on 5.1, luajit)

**Comment by isage** [comment]
What about dependency on libbson?

Also, is it tested? Last time I've tried I've encountered this bug https://github.com/luarocks/luarocks/issues/437

**Comment by bzick** [comment]
> What about dependency on libbson?

Good question. Now cmake tells me that libbson required when running `luarocks install`. But i think dependencies may be checked using [external_deps_patterns](https://github.com/luarocks/luarocks/wiki/Platform-agnostic-external-dependencies). I'll try it


> Also, is it tested? Last time I've tried I've encountered this bug luarocks/luarocks#437

Now it work fine. I'm already use rockspec (from own repo) in my project. But yesterday i caught a bug: luarocks always install lua-cbson into global tree, instead of local tree (when local package tree configured). I think it is problem in `LUA_LIBRARY = "$(LUA_LIBDIR)/lua"`. I'll try fix it. Please do not merge this request yet

**Comment by bzick** [comment]
Juts for test
```
luarocks install https://raw.githubusercontent.com/bzick/lua-cbson/master/rockspec/lua-cbson-git-1.rockspec
```

**Comment by isage** [comment]
>luarocks always install lua-cbson into global tree, instead of local tree

https://github.com/luarocks/luarocks/issues/432

File: rockspec/lua-cbson-git-1.rockspec (hunk 1 above)
Comment (bzick) [reply rockspec/lua-cbson-git-1.rockspec]: Fix in https://github.com/bzick/lua-cbson/commit/3606e4ba7956dca14b9774138cceacf097a97701

**Comment by bzick** [comment]
Fix local install: https://github.com/bzick/lua-cbson/commit/3606e4ba7956dca14b9774138cceacf097a97701 (seems like hack but works fine)

tested on macos and linux for local, custom and global install

**Pull request merged**